### PR TITLE
fix(datepicker): no longer trigger onChange when the value is equivalent

### PR DESCRIPTION
**Pull request by divdavem**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

File: src/datepicker/datepicker.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ export class NgbDatepicker implements OnDestroy,
     OnChanges, OnInit, ControlValueAccessor {
   model: DatepickerViewModel;
 
+  private _controlValue: NgbDate;
```
Comment (maxokorokov) [inline src/datepicker/datepicker.ts]: Do you mind if we move this to the datepicker service?

Not sure it it makes a huge difference, but the idea was to keep any state away from component.

I would have added an option:

```ts
this._service.select(..., {fromControl: true});
```

in this case the service's `select` would to something like:

```ts
if (isChangedDate(this._state.selectedDate, selectedDate)) {
        this._nextState({selectedDate, controlValue: options.fromControl ? selectedDate : undefined});
      }
```

P.S. names `formControl` and `controlValue` are just examples

Comment (divdavem) [reply src/datepicker/datepicker.ts]: Thank you for your comment! It is a good idea! I will update my PR.

Comment (divdavem) [reply src/datepicker/datepicker.ts]: Well, in fact, as discussed, I think it is better as it is now (as we need to update `controlValue` again when calling `onChange`).

**Comment by divdavem** [comment]
@maxokorokov Thank you for your review!

File: src/datepicker/datepicker.ts (hunk 1 above)
Comment (maxokorokov) [reply src/datepicker/datepicker.ts]: Yep, as discussed, simpler to leave it as is

**Pull request merged**